From bc29f98f0137fa1083a4cacf832d52f740d150a8 Mon Sep 17 00:00:00 2001
From: Jason Wessel <jason.wessel@windriver.com>
Date: Thu, 29 Jun 2017 20:33:23 -0700
Subject: [PATCH 5/8] Python3 compatibility: fix integer problems

Commit fa145f1a53943243f94a32ce98525db8494b0052 from
https://github.com/openvswitch/ovs.git

In python3 maxint is not defined, but maxsize is defined in both
python2 and python3.

The put_text() will not automatically use a value which came in as
float due to a pior math function and python3 will throw an exception.
The simple answer is to convert it with int() and move on.

Signed-off-by: Jason Wessel <jason.wessel@windriver.com>
Signed-off-by: Ben Pfaff <blp@ovn.org>
---
 ovsdb/ovsdb-idlc.in   | 2 +-
 python/build/nroff.py | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/ovsdb/ovsdb-idlc.in b/ovsdb/ovsdb-idlc.in
index 615548f..7cbcbf5 100755
--- a/ovsdb/ovsdb-idlc.in
+++ b/ovsdb/ovsdb-idlc.in
@@ -358,7 +358,7 @@ static void
                     print("        %s" % type.value.initCDefault(valueVar, type.n_min == 0))
                 print("    }")
             else:
-                if type.n_max != sys.maxint:
+                if type.n_max != sys.maxsize:
                     print("    size_t n = MIN(%d, datum->n);" % type.n_max)
                     nMax = "n"
                 else:
diff --git a/python/build/nroff.py b/python/build/nroff.py
index c23837f..401f699 100644
--- a/python/build/nroff.py
+++ b/python/build/nroff.py
@@ -148,6 +148,8 @@ def fatal(msg):
 
 
 def put_text(text, x, y, s):
+    x = int(x)
+    y = int(y)
     extend = x + len(s) - len(text[y])
     if extend > 0:
         text[y] += ' ' * extend
-- 
2.5.0

